File: chor-extraction/src/main/java/ast/cc/interfaces/CCNode.java
package ast.cc.interfaces;

import ast.cc.CCVisitor;
import epp.MergingException;

/**
 * Created by fmontesi on 03/04/17.
 */
public interface CCNode {
    <T> T accept( CCVisitor<T> visitor ) throws MergingException;
}

File: chor-extraction/src/main/java/ast/cc/interfaces/CCNode.kt
package ast.cc.interfaces

import ast.cc.CCVisitor
import epp.MergingException

/**
 * Created by fmontesi on 03/04/17.
 */
interface CCNode {
    @Throws(MergingException::class)
    fun <T> accept(visitor: CCVisitor<T>): T
}
